src/m365/aad/commands/o365group/o365group-recyclebinitem-restore.ts

```diff
@@ -29,29 +33,83 @@ class AadO365GroupRecycleBinItemRestoreCommand extends GraphCommand {
     return [commands.O365GROUP_RESTORE];
   }
 
-  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+  public async commandAction(logger: Logger, args: CommandArgs, cb: () => void): Promise<void> {
```

Comment:
`commandAction` is not an async method so we shouldn't change its signature because that could lead to undesired behavior in the CLI runtime.

Reply:
All right. Is it ok if I create an async function in the `commandAction` method?
```ts
public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
   (async () => {
      try {
        // Do the magic
        cb();
      }
      catch(err: any) {
        this.handleRejectedODataJsonPromise(err, logger, cb);
      }
    })();
}
```

Or should I use a promise -> then structure?

Reply:
Promise would be clearer IMHO. If I'm not mistaken, we have a separate issue regarding moving away from callbacks to Promises so we can look at it then.